src/carnot/funcs/builtins/pprof_ops.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <absl/container/flat_hash_map.h>
+
+#include <string>
+
+#include "src/carnot/udf/registry.h"
+#include "src/carnot/udf/udf.h"
+#include "src/shared/pprof/pprof.h"
+
+namespace px {
+namespace carnot {
+namespace builtins {
+
+using px::shared::PProfProfile;
+
+class CreatePProfRowAggregate : public udf::UDA {
+ public:
+  static udf::UDADocBuilder Doc() {
+    return udf::UDADocBuilder("Convert perf profiling data to pprof format.")
+        .Details("Converts perf profiling stack traces into pprof format.")
+        .Example(
+            R"example(stack_traces = px.DataFrame(table='stack_traces.beta', start_time='-1m')
+sample_period = px.GetProfilerSamplingPeriodMS()
+stack_traces.asid = px.asid()
+df = stack_traces.merge(sample_period, how='inner', left_on=['asid'], right_on=['asid'])df = df.groupby(['profiler_sampling_period_ms'])
```

Comment:
missing a newline between the merge and the groupby.

also we canonically chain the groupby and agg statements, i.e. `df = df.groupby(...).agg(...)`

Reply:
Also the example string is processed and the formatting is changed.
I believe you need to include the pipe char to include newlines in the generated doc, see `QuantilesUDA::Doc` for example.

Reply:
> also we canonically chain the groupby and agg statements, i.e. `df = df.groupby(...).agg(...)`

I had them split to avoid a long line. Will experiment with the formatting using the chained groupby + agg.

Reply:
> missing a newline between the merge and the groupby.

yikes, good catch, ty.